web/src/lib/utils/storage-helpers.ts

```diff
@@ -0,0 +1,146 @@
+import { RegistrationStatus, UserData } from "@businessnjgovnavigator/shared";
+
+type StorageType = "session" | "local";
+export type UseStorageReturnValue = {
+  get: (key: string, type?: StorageType) => string | undefined;
+  set: (key: string, value: string, type?: StorageType) => boolean;
+  keys: (type?: StorageType) => string[];
+  delete: (key: string, type?: StorageType) => void;
+  clear: (type?: StorageType) => void;
+};
+
+const UseStorage = (): UseStorageReturnValue => {
+  const storageType = (type?: StorageType): "localStorage" | "sessionStorage" =>
+    `${type ?? "session"}Storage`;
+
+  const isBrowser: boolean = ((): boolean => typeof window !== "undefined")();
+
+  const get = (key: string, type?: StorageType): string | undefined => {
+    return isBrowser ? window[storageType(type)].getItem(key) ?? undefined : undefined;
+  };
+
+  const set = (key: string, value: string, type?: StorageType): boolean => {
+    if (isBrowser) {
+      window[storageType(type)].setItem(key, value);
+      return true;
+    }
+    return false;
+  };
+
+  const keys = (type?: StorageType) =>
+    [...Array(window[storageType(type)].length)].map((_, i) => window[storageType(type)].key(i)) as string[];
+
+  const clear = (type?: StorageType) => window[storageType(type)].clear();
+
+  const _delete = (key: string, type?: StorageType): void => {
+    window[storageType(type)].removeItem(key);
+  };
+
+  return {
+    get,
+    set,
+    keys,
+    clear,
+    delete: _delete,
+  };
+};
+
+interface UserDataStorageReturnValue {
```

Comment:
can we separate these two objects into individual files? UseStorage is generic, whileUserDataStorage is specific to user data. 

Also, same comment as before, I'd prefer to not use the `Use`... prefix for things that aren't react hooks